src/main/java/org/hpccsystems/jdbcdriver/HPCCConnection.java

```diff
@@ -72,25 +76,42 @@ public class HPCCConnection implements Connection
     private String               catalog = HPCCJDBCUtils.HPCCCATALOGNAME;
 
     private Platform                    hpccPlatform              = null;
-    private HPCCWsClient                wsClient                  = null;
     private HPCCWsSQLClient             wsSQLClient               = null;
-    private String                      serverAddress;
-    private int                         wsEclPort;
+
     private String                      targetcluster;
     private String                      queryset;
     private String                      userName;
     private int                         pageSize;
     private int                         pageOffset;
     private int                         connectTimeoutMillis;
     private int                         readTimoutMillis;
-    private boolean                     hasTargetWsSQLBeenReached = false;
     private int                         eclResultLimit;
+    private boolean                     hasTargetWsSQLBeenReached = false;
 
     public HPCCConnection(Properties props)
     {
         this.connectionProps = props;
 
-        this.serverAddress = props.getProperty("ServerAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+        String wsECLWatchAddress = props.getProperty("WsECLWatchAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+        String wsSQLAddress = props.getProperty("ServerAddress", HPCCDriver.SERVERADDRESSDEFAULT);
+
+        if (wsSQLAddress.isEmpty())
+        {
+            SQLWarning warn = new SQLWarning("ServerAddress (WsSQL) not provided!");
+            addWarning(warn);
+            HPCCJDBCUtils.traceoutln(Level.SEVERE, warn.getMessage());
+
+            return;
+        }
+
+        if (wsECLWatchAddress.isEmpty())
+        {
+            wsECLWatchAddress = wsSQLAddress;
+            SQLWarning warn = new SQLWarning("WsECLWatch configuration not provided, WsSQL and ECLWatch expected on: '" + wsSQLAddress + "'");
+            addWarning(warn);
```

Comment:
Should this behave similar to the wsSQLAddress check and bail if the address isEmpty()?

Reply:
Maybe, but the assumption is that Wssql is not empty, and eclwatch would be accessible on that same address